sklearn/multioutput.py

```diff
@@ -404,8 +404,18 @@ def fit(self, X, Y, sample_weight=None, **fit_params):
         self.classes_ = [estimator.classes_ for estimator in self.estimators_]
         return self
 
-    @property
-    def predict_proba(self):
+    def _check_predict_proba(self):
+        check_is_fitted(self)
+        if not all(
+            [hasattr(estimator, "predict_proba") for estimator in self.estimators_]
```

Comment:
I see regarding the `check_is_fitted`. What we can do is move `check_is_fitted` in the `predict_proba` and the check here should check either the fitted `estimators_` if they exist, otherwise it should be the constructor attribute `estimator`. It could allow to know if the `predict_proba` is existing before calling `fit`. It is indeed one of the reason why we created `available_if` :)

Reply:
I still think that we should check for the unfitted estimator when `estimators_` is not available, similarly to what we try there: https://github.com/scikit-learn/scikit-learn/pull/20593/files#diff-d34a4e3caabf8c8864ac124708eb3bb65552006ae8d3f7b8f3b655c6c3557b60R53

Reply:
@glemaitre Thanks for the comment. `_check_predict_proba` should look like this?

```python
    def _check_predict_proba(self):
        if (
            not all(
                [hasattr(estimator, "predict_proba") for estimator in self.estimators_]
            )
            if hasattr(self, "estimators_")
            else check_is_fitted(self)
        ):
            raise AttributeError(
                "The base estimator should implement predict_proba method"
            )
        return True
```

Reply:
```python
err_msg = ...
if hasattr(self, "estimators_") and not all(
        [hasattr(estimator, "predict_proba") for estimator in self.estimators_]
):
    raise AttributeError(err_msg)
elif not hasattr(self, "estimator"):
    raise AttributeError(err_msg)
return True
```